presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ApplyNodeUtil.java

```diff
@@ -29,11 +34,21 @@ private ApplyNodeUtil() {}
     public static void verifySubquerySupported(Assignments assignments)
     {
         checkArgument(
-                assignments.getExpressions().stream().map(OriginalExpressionUtils::castToExpression).allMatch(ApplyNodeUtil::isSupportedSubqueryExpression),
+                assignments.getExpressions().stream().allMatch(ApplyNodeUtil::isSupportedSubqueryExpression),
                 "Unexpected expression used for subquery expression");
     }
 
-    public static boolean isSupportedSubqueryExpression(Expression expression)
+    private static boolean isSupportedSubqueryExpression(RowExpression expression)
+    {
+        if (isExpression(expression)) {
+            return isSupportedSubqueryExpression(castToExpression(expression));
+        }
+        return expression instanceof InSubqueryRowExpression ||
```

Comment:
Do you expect this to be different than checking instance of IntermediateFormRowExpression?

Reply:
Yes, eventually there will be more types of IntermediateFormRowExpression for different types of rules

Reply:
In that case, should we rename the plan checker to `verifyNoIntermediateFormExpression` since that's what it actually checks and also the error message there?